src/Core/Scripting/OOJSStation.m

```diff
@@ -947,31 +947,38 @@ static JSBool StationAddShipToShipyard(JSContext *context, uintN argc, jsval *vp
 		return NO;
 	}
 
+	// make sure the station has a shipyard
+	if ([station hasShipyard] == false) {
+		OOJSReportWarningForCaller(context, @"Station", @"removeShipFromShipyard", @"Station does not have shipyard.");
+		return NO;
+	}
 	// make sure the shipyard has been generated
 	if ([station localShipyard] == nil) [station generateShipyard];
 	NSMutableArray *shipyard = [station localShipyard];
 
 	if (JSVAL_IS_NULL(OOJS_ARGV[0]))  OOJS_RETURN_VOID;	// OK, do nothing for null ship.
 
-	NSMutableDictionary *shipyardDefinition = OOJSNativeObjectFromJSObject(context, JSVAL_TO_OBJECT(OOJS_ARGV[0]));
+	NSMutableDictionary *result = [NSMutableDictionary dictionary];
+	NSDictionary *shipyardDefinition = OOJSNativeObjectFromJSObject(context, JSVAL_TO_OBJECT(OOJS_ARGV[0]));
 	// validate each element of the dictionary
 	if (shipyardDefinition == nil) 
 	{
 		OOJSReportBadArguments(context, @"Station", @"addShipToShipyard", MIN(argc, 1U), OOJS_ARGV, nil, @"valid dictionary object");
 		return NO;
 	}
-	if ([shipyardDefinition objectForKey:@"short_description"] == nil) 
+	if ([shipyardDefinition objectForKey:KEY_SHORT_DESCRIPTION] == nil) 
```

Comment:
Same as before, stylewise I would prefer
`if (![shipyardDefinition objectForKey:KEY_SHORT_DESCRIPTION])` 
for consistency.